### ISSUE TITLE
Right-clicking on extension in statusbar and then "Manage extension" yields an "e.replace is not a function" error

### ISSUE THREAD

**Issue by V-ed** [Microsoft/vscode#55921]
Issue Type: <b>Bug</b>

This happens on every extension that has an item in the statusbar (might be obvious as the "Manage extension" menu item is a VSCode feature but worthy of pointing out nonetheless).

Interestingly, in the Insider's version, this error is not thrown, but nothing happens.

I'm expecting to go to the extension's homepage to review its available settings for example. I believe that was the behavior before, but I can't tell in which version was this Manage Extension menu item still working.

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:43:53.668Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.89GB (8.20GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (60)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|1.2.6
Bookmarks|ale|9.0.3
project-manager|ale|8.0.0
copy-relative-path|ale|0.0.2
vscode-intelephense-client|bme|0.8.8
better-phpunit|cal|0.8.1
better-whitespace|chm|1.0.1
gitignore|cod|0.5.0
bracket-pair-colorizer|Coe|1.0.59
vscode-svgviewer|css|1.4.4
vscode-markdownlint|Dav|0.19.0
vscode-quick-select|dba|0.2.6
vscode-faker|dee|1.3.0
githistory|don|0.4.2
jquerysnippets|don|0.0.1
ghlink|dt|1.0.2
gitlens|eam|8.5.4
vscode-html-css|ecm|0.2.0
EditorConfig|Edi|0.12.4
pull-request-monitor|eri|1.6.1
php-debug|fel|1.12.3
vsc-travis-ci-status|fel|1.1.0
auto-close-tag|for|0.5.6
auto-rename-tag|for|0.0.15
gc-excelviewer|Gra|2.1.25
vscode-auto-open-markdown-preview|hnw|0.0.4
crane|Hvy|0.3.8
output-colorizer|IBM|0.1.2
autofilename|Jer|0.1.3
vscode-random|jre|1.3.2
vscode-github|Kni|0.30.0
expand-region|let|0.1.2
vs-color-picker|lih|0.7.0
php-namespace-resolver|Meh|1.1.0
dotenv|mik|1.0.1
vscode-todo-parser|min|1.9.1
vscode-duplicate|mrm|1.2.1
sublime-keybindings|ms-|4.0.0
color-highlight|nau|2.3.0
php-docblocker|nei|1.6.0
snippet-creator|nik|0.0.4
laravel-blade|one|1.18.0
laravel5-snippets|one|1.4.0
vscode-php-getters-setters|php|1.2.3
material-icon-theme|PKi|3.5.2
polacode|pnp|0.2.2
vscode-css-peek|pra|2.1.0
view-in-browser|qin|0.0.5
java|red|0.29.0
LiveServer|rit|5.1.1
vscode-format|rya|0.0.6
code-settings-sync|Sha|3.0.0
gitconfig|sid|1.0.1
selectline-statusbar|tom|0.0.2
vscode-input-sequence|tom|0.2.0
vscode-java-debug|vsc|0.11.0
vscode-maven|vsc|0.10.0
better-align|wwm|1.1.6
licenser|ymo|1.1.2
markdown-all-in-one|yzh|1.6.0

(1 theme extensions excluded)

</details>

**Comment by ramya-rao-a** [comment]
I am able to use the `Manage Extension` menu item to open a editor tab for the extension and see all the extension details both in Stable (1.25.1) and in the latest Insiders.

Can you tell us which extension did you try? I used https://marketplace.visualstudio.com/items?itemName=ramyaraoa.show-offset

**Comment by V-ed** [comment]
This issue also happens on my side with your extension (`show-offset`), but here's those I've also tried :
- [`GitHub Pull Request Monitor`](https://marketplace.visualstudio.com/items?itemName=erichbehrens.pull-request-monitor)
- [`TODO Parser`](https://marketplace.visualstudio.com/items?itemName=minhthai.vscode-todo-parser)
- [`Travis CI Status`](https://marketplace.visualstudio.com/items?itemName=felixrieseberg.vsc-travis-ci-status)

---

I'm guessing that if it doesn't happen on your side, my installation is somewhat affected - be it an extension that hooks the `Manage Extension` or something else that I'm not aware on.

I'll try to see if this happen on my Linux partition, I'll get back here soon enough.

**Comment by V-ed** [comment]
> I'll try to see if this happen on my Linux partition, I'll get back here soon enough.

This issue is not present on a very similar installation on my Linux Mint boot image.

I hate saying "very similar" though, I'll edit this message once I'll be able to test the exact same configuration on another machine.

**Comment by V-ed** [comment]
@ramya-rao-a I think I found the culprit : I turned off the `"window.titleBarStyle": "custom"` setting and realized that the `Manage Extension` now works as supposed to.

Tried this in Insiders too (latest version as I'm writing this update) and the same happens : works **without** custom titleBar, fails (with no error in insiders) **with** custom titleBar.

Can you test it with `"window.titleBarStyle": "custom"` to see if you get this issue on your machine too?

Here's a log of the dev console's output in version `1.25.1` (I tried to click on manage extension twice) : 
[manageExtensionConsoleLog.log](https://github.com/Microsoft/vscode/files/2281149/manageExtensionConsoleLog.log) - in the case that this could help.

**Comment by ramya-rao-a** [comment]
Good catch @V-ed, I can confirm that the `Manage Extension` menu option doesnt work when that setting is set to `custom`.

**Comment by sbatten** [comment]
@bpasero In changing this I noticed that there is differing behavior between the custom context menu service and the native context menu service with regards to handling `getActionsContext`. In the native service, this isn't called until the action is run and passes the event into the handler. In the custom service, the handler is called on creation of the menu as it takes a context, not a function. Should the custom menu be updated to take the handler instead? I don't see anything in the code base that is actually passing a handler that uses the event, so I don't know if there is any value in passing it along. It seems the context menu handler just appends the event to the context object. 

```
export interface IContextMenuDelegate {
        ...
	getActionsContext?(event?: IEvent): any;
        ...
}
```

Native handling
```
	private runAction(actionRunner: IActionRunner, actionToRun: IAction, delegate: IContextMenuDelegate, event: IEvent): void {
		/* __GDPR__
			"workbenchActionExecuted" : {
				"id" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" },
				"from": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
			}
		*/
		this.telemetryService.publicLog('workbenchActionExecuted', { id: actionToRun.id, from: 'contextMenu' });

		const context = delegate.getActionsContext ? delegate.getActionsContext(event) : event;
		const res = actionRunner.run(actionToRun, context) || TPromise.as(null);

		res.done(null, e => this.notificationService.error(e));
	}
```

Custom handling
```
					const menu = new Menu(container, actions, {
						actionItemProvider: delegate.getActionItem,
						context: delegate.getActionsContext ? delegate.getActionsContext() : null,
						actionRunner,
						getKeyBinding: delegate.getKeyBinding
					});
```

**Comment by bpasero** [comment]
@sbatten yeah I think the important bit is to pass the context along to the action that gets executed, but I am not sure the event is used anywhere. My guess would be that it is maybe (or was maybe) used to find out what keys are pressed or what mouse button was used when invoking the action. We might have used this at one point but no longer and if so should maybe remove the event altogether.

**Comment by ramya-rao-a** [comment]
@sbatten @bpasero August or July recovery?

**Comment by bpasero** [comment]
I think we can move this to August given it only reproduces with the custom titlebar enabled.

**Comment by V-ed** [comment]
The issue is closed but is related to #52854
